flow_py_sdk/client/entities.py

```
@@ -149,17 +151,40 @@
-        self.value: cadence.Event = json.loads(payload, object_hook=cadence_object_hook)
+
+        # Add error handling for deserializing payload
+        max_payload_size = 1 * 1024 * 1024  # 1 MB size limit for payload
+        if len(payload) > max_payload_size:
+            logging.warning(f"Payload too large for event {event_index}: {len(payload)} bytes. Skipping deserialization.")
+            self.value = None
+        else:
+            try:
+                # Attempt to decode the payload
+                self.value: cadence.Event = json.loads(payload, object_hook=cadence_object_hook)
+            except json.JSONDecodeError as e:
```

Ruff

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)